tests/deployment/test_client_server.py

```diff
@@ -97,172 +99,192 @@ def test_client_server_sklearn(
 
     # Running the simulation using a model that is not compiled should not be possible
     with pytest.raises(AttributeError, match=".* model is not compiled.*"):
-        client_server_simulation(x_train, x_test, model, default_configuration)
+        check_client_server_execution(
+            x_test, model, key_dir, check_array_equal, check_float_array_equal
+        )
+
+    # Compile the model
+    fhe_circuit = model.compile(x_train, configuration=default_configuration)
 
-    fhe_circuit = model.compile(x_train, default_configuration, **extra_params, show_mlir=False)
+    # Check that client and server files are properly generated
+    check_client_server_files(model)
 
     max_bit_width = fhe_circuit.graph.maximum_integer_bit_width()
     print(f"Max width {max_bit_width}")
 
-    # Compare the FHE predictions with the clear ones.
-    # Note that:
-    # - With a global_p_error of 1/100_000 we only allow one run.
-    # - The simulated predictions are not considered in this test.
+    # Compare the FHE predictions with the clear ones. Simulated predictions are not considered in
+    # this test.
     check_is_good_execution_for_cml_vs_circuit(x_test, model, simulate=False, n_allowed_runs=1)
 
     # Check client/server FHE predictions vs the FHE predictions of the dev model
-    client_server_simulation(x_train, x_test, model, default_configuration)
+    check_client_server_execution(
+        x_test, model, key_dir, check_array_equal, check_float_array_equal
+    )
 
 
 def test_client_server_custom_model(
-    default_configuration, check_is_good_execution_for_cml_vs_circuit
+    default_configuration,
+    check_is_good_execution_for_cml_vs_circuit,
+    check_array_equal,
+    check_float_array_equal,
 ):
-    """Tests the client server custom model."""
+    """Test the client-server interface for a custom model (through a quantized module)."""
 
     # Generate random data
     x_train, x_test = numpy.random.rand(100, 2), numpy.random.rand(1, 2)
 
+    key_dir = default_configuration.insecure_key_cache_location
+
     # Running the simulation using a QuantizedModule that is not compiled should not be possible
     with pytest.raises(AttributeError, match=".* quantized module is not compiled.*"):
         # Instantiate an empty QuantizedModule object
         quantized_module = QuantizedModule()
 
-        client_server_simulation(x_train, x_test, quantized_module, default_configuration)
+        check_client_server_execution(
+            x_test, quantized_module, key_dir, check_array_equal, check_float_array_equal
+        )
 
     torch_model = FCSmall(2, nn.ReLU)
-    n_bits = 2
 
-    # Get the quantized module from the model
+    # Get the quantized module from the model and compile it
     quantized_numpy_module = compile_torch_model(
         torch_model,
         x_train,
         configuration=default_configuration,
-        n_bits=n_bits,
-        global_p_error=1 / 100_000,
```

Comment:
I thought we wanted to limit the use of global_p_error, why do we have this here ?

Reply:
This is legacy code when we had global_p_error = 0.01 by default in CML. Great that you remove it. Not sure about the n_bits though.

Reply:
I see thanks ! regarding the n_bits, I just removed the variable and directly put `n_bits=2` in the `compile` method as it wasn't used anywhere else